packages/widgetbook_core/lib/src/navigation_tree/widgets/navigation_tree_item.dart

```diff
@@ -0,0 +1,165 @@
+import 'package:flutter/material.dart';
+import 'package:widgetbook_core/widgetbook_core.dart';
+
+class NavigationTreeItem extends StatefulWidget {
+  const NavigationTreeItem({
+    super.key,
+    this.level = 0,
+    this.label,
+    this.content,
+    this.iconData,
+    this.icon,
+    this.onTap,
+    this.isExpanded = false,
+    this.onMoreIconPressed,
+  })  : assert(
+          label != null || content != null,
+          'Either a label or content must be provided',
+        ),
+        assert(
+          icon != null || iconData != null,
+          'Either an icon or iconData must be provided',
+        );
+
+  const NavigationTreeItem.category({
+    super.key,
+    required this.label,
+    this.onTap,
+    this.isExpanded = false,
+    this.onMoreIconPressed,
+  })  : level = 0,
+        iconData = Icons.auto_awesome_mosaic,
+        icon = null,
+        content = null;
+
+  const NavigationTreeItem.package({
+    super.key,
+    required this.label,
+    this.onTap,
+    this.isExpanded = false,
+    this.onMoreIconPressed,
+  })  : level = 0,
+        iconData = Icons.inventory,
+        icon = null,
+        content = null;
+
+  const NavigationTreeItem.folder({
+    super.key,
+    required this.label,
+    this.onTap,
+    this.isExpanded = false,
+    this.level = 0,
+    this.onMoreIconPressed,
+  })  : iconData = Icons.folder,
+        icon = null,
+        content = null;
+
+  const NavigationTreeItem.component({
+    super.key,
+    required this.label,
+    this.level = 0,
+    this.onTap,
+    this.isExpanded = false,
+    this.onMoreIconPressed,
+  })  : iconData = null,
+        icon = const ComponentIcon(),
+        content = null;
+
+  const NavigationTreeItem.useCase({
+    super.key,
+    required this.label,
+    this.level = 0,
+    this.onTap,
+    this.isExpanded,
+    this.onMoreIconPressed,
+  })  : iconData = null,
+        icon = const UseCaseIcon(),
+        content = null;
+
+  static const double indentation = 25;
```

Comment:
from a logical perspective I wasn't able to explain why this should be 25 (and not 24).. but it works :D

Reply:
"If it works ship it" 🚀 :)